feat: Add PDF thumbnail generation feature

princethakur931 opened the pull request:
## 📄 PDF Thumbnail Generation Feature

### Overview
This PR implements automatic PDF thumbnail generation for uploaded documents. Thumbnails are pre-generated during upload and displayed in the collection list view for faster loading and better visual representation.

### Changes Made

#### New Files
- **`app/Services/ThumbnailService.php`** - Core service for PDF thumbnail generation
  - Supports both Imagick and Ghostscript + GD library
  - Generates 200x200px JPEG thumbnails from first page
  - Stores at `storage/app/public/doc-thumbnails/{collection_id}/{document_id}/`
  
- **`app/Console/Commands/RegenerateThumbnails.php`** - Artisan command
  - Usage: `php artisan SR:RegenerateThumbnails {collection_id}` or `all`
  - Bulk regeneration with progress bar
  - Based on RebuildElasticIndex command structure

#### Modified Files
- **`app/Document.php`**
  - Added `getThumbnailUrl()` method - returns thumbnail URL or null
  - Added `hasThumbnail()` method - checks if thumbnail exists

- **`app/Http/Controllers/DocumentController.php`**
  - Integrated thumbnail generation in upload process
  - Auto-generates thumbnails for PDF uploads
  - Non-blocking (upload succeeds even if thumbnail fails)

- **`app/Traits/Search.php`**
  - Modified document list display logic
  - Shows PDF thumbnails when available
  - Falls back to default PDF icon if not present

- **`config/filesystems.php`**
  - Enabled `public` disk for thumbnail storage

- **`composer.json`**
  - Added `spatie/pdf-to-image` package (v3.1.0)

### Features
✅ **Automatic thumbnail generation** - Creates thumbnails during PDF upload  
✅ **Graceful fallback** - Uses Ghostscript + GD if Imagick unavailable  
✅ **Batch regeneration** - Command to regenerate entire collections  
✅ **Smart display** - Shows thumbnails in list view with fallback to icons  
✅ **Organized storage** - Clean directory structure per collection/document  
✅ **Non-blocking** - Upload succeeds even if thumbnail generation fails  

### Testing
- Tested with GD + Ghostscript combination
- Thumbnails generate successfully on upload
- `SR:RegenerateThumbnails` command working correctly
- List view displays thumbnails properly

### Requirements
- **PHP GD extension** (enabled) ✅
- **Ghostscript** (installed at `C:\Program Files\gs\`) ✅
- **Storage link** created (`php artisan storage:link`) ✅

### Usage
```bash
# Regenerate thumbnails for specific collection
php artisan SR:RegenerateThumbnails 2

# Regenerate thumbnails for all collections
php artisan SR:RegenerateThumbnails all
```

ScreenShot / Testing Output: 

<img width="1920" height="1080" alt="image" src="https://github.com/user-attachments/assets/f534a863-4a4b-442f-8eec-b5416d699a89" />

princethakur931 committed:
feat: Add PDF thumbnail generation feature

ketan404 commented on config/filesystems.php:
There should be no update to this file, as far as I can see. I believe public disk is already active. Besides, this depends on the default disk in the system. It's true that most of the times, it will storage/app.
The installation does create a symbolic link to the public disk (public/storage => storage/app/public)

ketan404 commented on app/Http/Controllers/DocumentController.php:
Please move this piece of code to the listener - app/Listener/DocumentSaved.

ketan404 commented on app/Traits/Search.php:
method_exists() call is not required. If you are convinced it is, call me.

ketan404 reviewed (changes requested):
Comments added

ketan404 commented on app/Listeners/DocumentSaved.php:
This is called in two scenarios.
1. When a new document is uploaded.
2. When a document (record) is edited. In this case, there may not be any file upload. Therefore, check if there a pdf file that is uploaded along with the other data. Regenerate icon only if true.

ketan404 commented on app/Services/ThumbnailService.php:
There will be too many directories if you create one dir for each thumbnail. Instead, create directories collection wise and the thumbnails could be saved by name - $doc->id."_thumb.jpg"

ketan404 reviewed (changes requested):
A couple more comments added.

princethakur931 committed:
Enable public disk for thumbnail storage

- Uncommented public disk configuration
- FILESYSTEM_DRIVER remains 'local' (default)
- Required for PDF thumbnail feature to store generated thumbnails

princethakur931 committed:
Optimize thumbnail generation and storage structure

- Only generate thumbnails when PDF file is actually uploaded
- Check file modification time vs document update time (5 min window)
- Changed directory structure: doc-thumbnails/{collection_id}/{document_id}_thumb.jpg
- Reduces directory count (one per collection instead of per document)
- Updated all references in ThumbnailService, Document model, and DocumentSaved listener
